pkg/streamingpromql/functions.go

```diff
@@ -65,56 +65,78 @@ func TransformationFunctionOperatorFactory(name string, seriesDataFunc functions
 // Returns:
 //
 //	An InstantVectorFunctionOperator.
-func LabelManipulationFunctionOperatorFactory(name string, metadataFunc functions.SeriesMetadataFunction) InstantVectorFunctionOperatorFactory {
+func InstantVectorLabelManipulationFunctionOperatorFactory(name string, metadataFunc functions.SeriesMetadataFunction) InstantVectorFunctionOperatorFactory {
 	return SingleInputVectorFunctionOperatorFactory(name, metadataFunc, functions.Passthrough)
 }
 
-func createRateFunctionOperator(args []types.Operator, pool *pooling.LimitingPool) (types.InstantVectorOperator, error) {
-	if len(args) != 1 {
-		// Should be caught by the PromQL parser, but we check here for safety.
-		return nil, fmt.Errorf("expected exactly 1 argument for rate, got %v", len(args))
-	}
+// SingleRangeVectorFunctionOperatorFactory creates an InstantVectorFunctionOperatorFactory for functions
+// that have exactly 1 argument (v range-vector).
+//
+// Parameters:
+//   - name: The name of the function.
+//   - metadataFunc: The function for handling metadata
+//   - rangeStepFunc: The function to handle a range vector step
+func SingleRangeVectorFunctionOperatorFactory(name string, metadataFunc functions.SeriesMetadataFunction, rangeStepFunc functions.RangeVectorStepFunction) InstantVectorFunctionOperatorFactory {
+	return func(args []types.Operator, pool *pooling.LimitingPool) (types.InstantVectorOperator, error) {
+		if len(args) != 1 {
+			// Should be caught by the PromQL parser, but we check here for safety.
+			return nil, fmt.Errorf("expected exactly 1 argument for %s, got %v", name, len(args))
+		}
 
-	inner, ok := args[0].(types.RangeVectorOperator)
-	if !ok {
-		// Should be caught by the PromQL parser, but we check here for safety.
-		return nil, fmt.Errorf("expected a range vector argument for rate, got %T", args[0])
+		inner, ok := args[0].(types.RangeVectorOperator)
+		if !ok {
+			// Should be caught by the PromQL parser, but we check here for safety.
+			return nil, fmt.Errorf("expected a range vector argument for %s, got %T", name, args[0])
+		}
+
+		return &operators.FunctionOverRangeVector{
+			Inner:               inner,
+			Pool:                pool,
+			MetadataFunc:        metadataFunc,
+			RangeVectorStepFunc: rangeStepFunc,
+		}, nil
 	}
+}
+
+// SingleRangeVectorTransformationFunctionOperatorFactory creates an InstantVectorFunctionOperatorFactory for functions
+// that have exactly 1 argument (v range-vector), and drop the series __name__ label.
+//
+// Parameters:
+//   - name: The name of the function.
+//   - rangeStepFunc: The function to handle a range vector step
 
-	return &operators.FunctionOverRangeVector{
-		Inner: inner,
-		Pool:  pool,
-	}, nil
+func SingleRangeVectorTransformationFunctionOperatorFactory(name string, rangeStepFunc functions.RangeVectorStepFunction) InstantVectorFunctionOperatorFactory {
```

Comment:
[nit] For consistency with `InstantVectorTransformationFunctionOperatorFactory`, should we call this `RangeVectorTransformationFunctionOperatorFactory`?